apps/web/src/components/Shared/Login/New/index.tsx

```diff
@@ -50,45 +58,78 @@ const NewProfile: FC<NewProfileProps> = ({ isModal = false }) => {
       }
     ],
     enabled: Boolean(debouncedHandle)
   });
-  const { data, write } = useContractWrite(config);
+  const { data: txData, write } = useContractWrite(config);
 
   const { isLoading, error } = useWaitForTransaction({
-    hash: data?.hash
+    hash: txData?.hash
   });
 
-  /*const handleUpload = async (evt: ChangeEvent<HTMLInputElement>) => {
-                                                                          evt.preventDefault();
-                                                                          setUploading(true);
-                                                                          try {
-                                                                            const attachment = await uploadToIPFS(evt.target.files);
-                                                                            if (attachment[0]?.item) {
-                                                                              setAvatar(attachment[0].item);
-                                                                            }
-                                                                          } finally {
-                                                                            setUploading(false);
-                                                                          }
-                                                                        };*/
+  const handleUpload = async (evt: ChangeEvent<HTMLInputElement>) => {
+    evt.preventDefault();
+    setUploading(true);
+    try {
+      const attachment = await uploadToIPFS(evt.target.files);
+      if (attachment[0]?.item) {
+        setAvatar(attachment[0].item);
+      }
+    } finally {
+      setUploading(false);
+    }
+  };
 
   const handleCreateProfile = async () => {
     write?.();
   };
 
-  return (
-    <Form form={form} className="space-y-4" onSubmit={handleCreateProfile}>
-      {error && (
+  useEffect(() => {
+    setIsCreationLoading(
+      IS_RELAYER_AVAILABLE ? data?.createProfile.__typename === 'RelayerResult' : !!txData?.hash
+    );
+  }, [data, txData]);
+
+  return isCreationLoading ? (
+    <Pending
+      handle={form.getValues('handle')}
+      txHash={
+        IS_RELAYER_AVAILABLE && data?.createProfile.__typename === 'RelayerResult'
+          ? data?.createProfile?.txHash
+          : txData?.hash
+      }
+    />
+  ) : (
+    <Form
+      form={form}
+      className="space-y-4"
+      onSubmit={({ handle }) => {
+        if (IS_RELAYER_AVAILABLE) {
+          const username = handle.toLowerCase();
+          createProfile({
+            variables: {
+              request: {
+                handle: username,
+                profilePictureUri: avatar ? avatar : getStampFyiURL(address ?? ZERO_ADDRESS)
+              }
+            }
+          });
+        } else {
+          handleCreateProfile();
+        }
+      }}
+    >
+      {data?.createProfile.__typename === 'RelayError' && (data?.createProfile.reason || error) && (
```

Comment:
I'm not sure we need to check for a reason as we use a condition for the message?
This should be sufficient then:
```suggestion
      {data?.createProfile.__typename === 'RelayError' && (
```

Reply:
As discussed, we keep it like it was done initially